tests/backend/loadSettings.js

```diff
@@ -7,20 +7,10 @@
 
 var jsonminify = require(__dirname+"/../../src/node_modules/jsonminify");
 const fs = require('fs');
+var settings = require("../../src/node/utils/Settings");
 
 function loadSettings(){
-  var settingsStr = fs.readFileSync(__dirname+"/../../settings.json").toString();
```

Comment:
So this code was bugged since forever, isnt'it?

If you read at the top of this file

https://github.com/ether/etherpad-lite/blob/c71706fd86c8e91323ff731f2b472179cb01a486/tests/backend/loadSettings.js#L1-L6

There is a "cloned" version in `tests/container/loadSettings.js`. Until the code is unified, we have to manually port the same modifications there. But: the previous (bugged) code "apparently" gave the possibility to explicitly load a settings file, and in `tests/container/loadSettings.js`, we were using it:

https://github.com/ether/etherpad-lite/blob/83d72d27a405f63782803e0168b1e9121f3cd650/tests/container/loadSettings.js#L12

See that we are loading `settings.json.docker`?

What happens when we port this modification to that file? What was the settings bug about?

Reply:
The settings bug was about if you ran ``loadSettings`` it only got the settings from the ``settings.json`` file, not the default settings IE after being processed by ``Settings.js``.

And yes this code was bugged since forever, the reason being is that if you require("Settings"); it has to init DB etc.  For tests it's probably fine to use ``settings.json`` but I wouldn't say it's ideal because your test settings might !== settings being loaded by ``Settings.js``

Reply:
After removing all the unused imports (1b300242400e) this is what remains of `tests/backend/loadSettings.js`:

https://github.com/ether/etherpad-lite/blob/1b300242400e9c7b664a5b84985a230f5736a4ec/tests/backend/loadSettings.js#L8-L14

I say there is no point in keeping it. **The backend test specs can directly require `src/node/utils/Settings`**.

There is a problem, though: the `loadSettings` mechanism was put in place to be able to live patch a configuration object, we discovered it never worked, so we are discarding it, but it is needed for testing the docker build.

We'll have to re-implement it, and all the Settings class is ging to need to be reworked. :sob: Basically we'll have to get ride of that huge side effect performed at require() time (parses the cli, reads files, dynamically modifies its exports... all implicitly).

Reply:
Yep!

Reply:
What's the conclusion here?  Anything required by me?

Reply:
> What's the conclusion here?

- I am going to drop `etherpad-lite/tests/backend/loadSettings.js`. All the backend tests will directly require Settings, since that's the intended behaviour anyhow;
- I am letting `tests/container/loadSettings.js` live as it was, because it served its purpose, and it is going to be cahnged once Settings is reimplemented as a module that does not do heavy side effects on load

> Anything required by me?
Not here. See my next point on UI when the rate limiter triggers.